shipyard-controller/db/mongodb_event_repo.go

```diff
@@ -76,6 +76,24 @@ func (mdbrepo *MongoDBEventsRepo) GetEvents(project string, filter common.EventF
 	return events, nil
 }
 
+// GetEventByID gets a single event of a project, based on the id
+func (mdbrepo *MongoDBEventsRepo) GetEventByID(projectName string, filter common.EventFilter, status ...common.EventStatus) (*apimodels.KeptnContextExtendedCE, error) {
+	collection, ctx, cancel, err := mdbrepo.getEventsCollection(projectName, status...)
+
+	if err != nil {
+		return nil, err
+	}
+
+	defer cancel()
+
+	cur := collection.FindOne(ctx, getSearchOptions(filter))
+
+	var event apimodels.KeptnContextExtendedCE
+	cur.Decode(&event)
+
+	return &event, cur.Err()
```

Comment:
NIT: I'd keep it consistent with the rest of the methods
```suggestion
	return event, cur.Err()
```